src/librustc/mir/mod.rs

```diff
@@ -1761,7 +1761,7 @@ impl_stable_hash_for!(struct Static<'tcx> {
     Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable, HashStable,
 )]
 pub struct Projection<'tcx> {
-    pub base: Place<'tcx>,
+    pub base: Option<Box<Projection<'tcx>>>,
```

Comment:
It is somewhat confusing to have `base` sometimes be a `Projection` and sometimes a `PlaceBase`.

Reply:
Yes it is, but the entire Projection type will be gone soon, so I deemed it irrelevant